phpBB/assets/javascript/installer.js

```diff
@@ -0,0 +1,270 @@
+/**
+ * Installer's AJAX frontend handler
+ */
+
+(function($) { // Avoid conflicts with other libraries
+    // Installer variables
+    var pollTimer = null;
+    var nextReadPosition = 0;
+    var progressBarTriggered = false;
+    var progressTimer = null;
+    var currentProgress = 0;
+
+    // Template related variables
+    var $contentWrapper = $('.install-body').find('.main');
+
+    // Intercept form submits
+    intercept_form_submit($('#install_install'));
+
+    function poll_content(xhReq) {
+        var messages = xhReq.responseText;
+
+        do {
+            var unprocessed = messages.substring(nextReadPosition);
+            var messageEndIndex = unprocessed.indexOf('}\n\n');
```

Comment:
It's fine, JavaScript has function scoping, not block scoping.

Having said that, it duplicates the variable declaration because the variables are defined in the loop. Won't cause an error to be thrown or affect behaviour, but is still worng.